File: src/MARCH15/cnote.py
# http://www.codechef.com/MARCH15/problems/CNOTE

for terms in range(int(raw_input())):
    x, y, k, n = map(int, raw_input().split())
    req_pages = x - y
    flag = False
    for i in range(n):
        pi, ci = map(int, raw_input().split())
        if pi >= req_pages and ci <= k:
            flag = True
    if flag:
        print "LuckyChef"
    else:
        print "UnluckyChef"

File: src/MARCH15/cnote.c
#include <stdio.h>
int main (void){
    int terms;
    scanf("%d", &terms);
    while(terms--){
        int x, y, k, n;
        scanf("%d%d%d%d", &x, &y, &k, &n);
        int req_pages = x - y;
        int i, flag = 0;
        for(i=1;i<=n;i++){
            int pi, ci;
            scanf("%d%d", &pi, &ci);
            if( pi>=req_pages && ci<=k )
                flag = 1;
        }
        if(flag)
            printf("LuckyChef\n");
        else
            printf("UnluckyChef\n");
    }
}
